python/Scripts/genmdl.py

```diff
@@ -43,13 +43,41 @@ def _loadLibraries(doc, searchPath, libraryPath):
             filePath = os.path.join(libraryPath, os.path.join(path, filename))
             _loadLibrary(filePath, doc)
 
-def _writeHeader(file, version):    
+def _writeHeader(file, version):
     file.write('mdl ' + version + ';\n')
     IMPORT_LIST = { 'anno', 'base', 'df', 'math', 'state', 'tex' }
     # To verify what are the minimal imports required
     for i in IMPORT_LIST:
         file.write('import::' + i + '::*;\n')
     file.write('\n')
+    file.write('// Custom color4 type\n')
+    file.write('export struct color4 {\n')
+    file.write('  color3 rgb  = color3(0.0);\n')
+    file.write('  float  a    = 1.0;\n')
+    file.write('};\n\n')
+
+def _mapGeomProp(geomprop):
+    outputValue = ''
+    if len(geomprop):
+        if geomprop.find('UV') >= 0:
+            outputValue = '::state::texture_coordinate(0).x, ::state::texture_coordinate(0).y'
+        elif geomprop.find('Pobject') >= 0:
+            outputValue = '::state::position()'
+        elif geomprop.find('PWorld') >= 0:
+            outputValue = '::state::position()'
+        elif geomprop.find('Nobject') >= 0:
+            outputValue = '::state::normal()'
+        elif geomprop.find('Nworld') >= 0:
+            outputValue = '::state::normal()'
+        elif geomprop.find('Tobject') >= 0:
+            outputValue = '::state::texture_tangent_u(0)'
+        elif geomprop.find('Tworld') >= 0:
+            outputValue = '::state::texture_tangent_u(0)'
+        elif geomprop.find('Bobject') >= 0:
+            outputValue = '::state::texture_tangent_u(0)'
+        elif geomprop.find('Bworld') >= 0:
+            outputValue = '::state::texture_tangent_v(0)'
+    return outputValue
```

Comment:
I didn't know what to do about world/object space since there seems to no intrinsic for this.